code/mapping/ext_modules/mapping_common/map.py

```
@@ -229,6 +265,75 @@
+                and entry.flags._is_lanemark is False
+            ):
+                return 1
+
+        # decrease buffer size for lane checkking
+        local_trajectory_buffer = local_trajectory.buffer(0.5, 3)
+        # Check left lane
+        left_lane = self.lane_marking_left()
+        if left_lane is not None:
+            if shapely.intersection(
+                left_lane.to_shapely().poly, local_trajectory_buffer
+            ):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)